src/main/java/liqp/Template.java

```diff
@@ -589,6 +596,53 @@ public String renderUnguarded(TemplateContext parent) {
         return renderUnguarded(new HashMap<String, Object>(), parent, true);
     }
 
+    /**
+     * Returns the processed placeholder input string post parsing and rending.
+     *
+     * @param inputString: input string that contains 0 or multiple pre-processed placeholder strings.
+     * @param variables: contains the mapping of the placeholders and is used for
+     *                 mapping of the placeholder strings while rendering the input string.
+     */
+    public static String processPlaceHolderString(String inputString, Map<String, Object> variables) {
+        StringBuilder result = new StringBuilder();
+
+        int lastMatchEnd = 0;
+        Matcher matcher = null;
+        Pattern pattern;
+
+        try {
+            pattern = Pattern.compile("\\{\\{[^{}]*\\}\\}");
+            matcher = pattern.matcher(inputString);
+        } catch (Exception exception) {
+            System.err.println("Returning empty string. Exception occurred while searching for parsing placeholder strings: " + exception.getMessage());
+            return "";
+        }
+
+        // Iterate through matches and modify them
+        while (Objects.nonNull(matcher) && matcher.find()) {
+            try {
+                result.append(inputString, lastMatchEnd, matcher.start());
+                String placeholder = matcher.group();
+
+                Template template = parse(placeholder);
+                String currentResultString = template.render(variables);
+
+                result.append(currentResultString);
+                lastMatchEnd = matcher.end();
+            } catch (Exception exception) {
+                String emptyString = new String();
+                result.append(emptyString);
+                lastMatchEnd = matcher.end();
+                System.err.println("Exception occurred while searching for parsing placeholder strings: " +
```

Comment:
extract all System.err to a separate method and call them with string passed unless we have logger added.
or add slf4j dependency and use that pls.